wasm/src/types/field.rs

```diff
@@ -14,27 +14,95 @@
 // You should have received a copy of the GNU General Public License
 // along with the Aleo SDK library. If not, see <https://www.gnu.org/licenses/>.
 
-use crate::types::native::FieldNative;
-
-use wasm_bindgen::prelude::wasm_bindgen;
+use crate::{
+    types::native::{FieldNative, LiteralNative, PlaintextNative, Uniform},
+    Plaintext,
+};
+use snarkvm_console::prelude::{Double, One, Pow};
+use std::ops::Deref;
 
+use once_cell::sync::OnceCell;
 use std::str::FromStr;
+use wasm_bindgen::prelude::wasm_bindgen;
 
 #[wasm_bindgen]
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct Field(FieldNative);
 
 #[wasm_bindgen]
 impl Field {
+    /// Creates a field object from a string representation of a field.
+    #[wasm_bindgen(js_name = "fromString")]
+    #[allow(clippy::should_implement_trait)]
+    pub fn from_string(field: &str) -> Result<Field, String> {
+        Ok(Self(FieldNative::from_str(field).map_err(|e| e.to_string())?))
+    }
+
+    /// Create a plaintext element from a group element.
+    #[wasm_bindgen(js_name = "toPlaintext")]
+    pub fn to_plaintext(&self) -> Plaintext {
+        Plaintext::from(PlaintextNative::Literal(LiteralNative::Field(self.0), OnceCell::new()))
+    }
+
+    /// Returns the string representation of the field.
     #[wasm_bindgen(js_name = "toString")]
     #[allow(clippy::inherent_to_string)]
     pub fn to_string(&self) -> String {
         self.0.to_string()
     }
 
-    #[wasm_bindgen(js_name = "fromString")]
-    pub fn from_string(field: &str) -> Result<Field, String> {
-        Ok(Self(FieldNative::from_str(field).map_err(|e| e.to_string())?))
+    /// Generate a random field element.
+    pub fn random() -> Field {
+        let rng = &mut rand::thread_rng();
+        Field(FieldNative::rand(rng))
+    }
+
+    /// Add two field elements.
+    pub fn add(&self, other: &Field) -> Field {
+        Field(self.0 + other.0)
+    }
+
+    /// Subtract two field elements.
+    pub fn subtract(&self, other: &Field) -> Field {
+        Field(self.0 - other.0)
+    }
+
+    /// Multiply two field elements.
+    pub fn multiply(&self, other: &Field) -> Field {
+        Field(self.0 * other.0)
+    }
+
+    /// Divide two field elements.
+    pub fn divide(&self, other: &Field) -> Field {
+        Field(self.0 / other.0)
+    }
+
+    /// Power of a field element.
+    pub fn pow(&self, other: &Field) -> Field {
+        Field(self.0.pow(other.0))
+    }
+
+    /// Invert the field element.
+    pub fn inverse(&self) -> Field {
```

Comment:
This should be called `negate`.

Reply:
Hmmmm, in that case I don't think it should be using the `-` operator... since `-` should have the meaning of "negate", not "invert".

Reply:
The `Neg` trait in is used in SnarkVM to implement inversions, so it's not something I can modify. The `negation` operator in Fields usually just means the inverse, but it generally assumes you're working in an integer ring or floats modeling the `Real Numbers` where there are so called "negative" numbers. Finite fields don't technically have negative numbers, but the negation symbol is used to denote inversions. 

The actual operation is called `inverse` though in Field theory and in SnarkVM so it's the name that needs to stay on this function.

Reply:
Thanks for the explanation, that helps a lot.